core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -356,6 +356,15 @@ trait OptimisticTransactionImpl extends TransactionalWrite
     updateMetadataInternal(_metadata, ignoreDefaultProperties)
   }
 
+  /**
+   * This updates the protocol for the table to a set newProtocol
+   */
+  def updateProtocol(newProtocolOption: Option[Protocol]): Unit = {
```

Comment:
i dont like this method signature. if someone is calling updateProtocol, then they must have a new protocol object. why would they pass on option? For example, updateMetadata just take metadata, not an option.